pandas/core/reshape/merge.py

```diff
@@ -1089,13 +1089,15 @@ def _validate_specification(self):
                     raise ValueError('len(left_on) must equal the number '
                                      'of levels in the index of "right"')
                 self.right_on = [None] * n
+            self.right_on = self.right_on or [None] * len(self.left_on)
         elif self.right_on is not None:
             n = len(self.right_on)
             if self.left_index:
                 if len(self.right_on) != self.left.index.nlevels:
                     raise ValueError('len(right_on) must equal the number '
                                      'of levels in the index of "left"')
                 self.left_on = [None] * n
+            self.left_on = self.left_on or [None] * len(self.right_on)
```

Comment:
you don’t need the lines above (not both cases)
also can use n

Reply:
not sure I get it. The previous line needs to go?

Reply:
try it - i think it’s unnecessary

Reply:
u can also combine the condition in 1095 and 1096 i think as well

Reply:
```left_index``` and ```right_index``` are ```False``` by default (in the _MergeOperation instantiation), so both left_on and right_on would need to be defined out of the if clause at 1095 and 1087 (correct me if I'm wrong)

Reply:
ok we’ll see if u can simplify things with passing tests